### ISSUE TITLE
Re-exporting Functions

### ISSUE THREAD

**Issue by chgibb** [AssemblyScript/assemblyscript#113]
Thanks again for creating such an amazing project. What I'm trying to accomplish is creating a single entry file for compilation which would include all exports from all relevant modules to allow for some form of code bundling between multiple ```.wasm``` files. My hope is to avoid direct compilation of individual, small modules to many ```.wasm``` files. Essentially, what we have in the Javascript world of bundling an entry point into one JS file, I wish to accomplish with Assemblyscript.

My attempt at this is as follows:

I have some module, called add with a single exported function.
![add ts test2 visual studio code_015](https://user-images.githubusercontent.com/18079458/30379390-53ba9cc4-9864-11e7-9e03-7fae9a3cd495.png)

As an attempt at an entry point, I have some module called main.
![main ts test2 visual studio code_015](https://user-images.githubusercontent.com/18079458/30379441-7faa8916-9864-11e7-9264-d918e7fc55ad.png)

My thought was to use Typescript's module re-exporting functionality to expose the exports of the imported modules. However, trying to compile main results in the following.
![compile](https://user-images.githubusercontent.com/18079458/30379489-a5625526-9864-11e7-9aac-8a3f754c1438.png)

**Comment by dcodeIO** [comment]
Yeah, re-exporting hasn't got much love, yet, but what you are trying to do is exactly how it is supposed to work once implemented.

**Comment by chgibb** [comment]
Could you point me in the right direction in the code base?  I could take a stab at it.

**Comment by chgibb** [comment]
@dcodeIO

**Comment by dcodeIO** [comment]
Resolving is currently done [here](https://github.com/AssemblyScript/assemblyscript/blob/master/src/compiler.ts#L1306). It's based on the assumption that TypeScript resolves all these things and links them to the correct [symbol](https://github.com/AssemblyScript/assemblyscript/blob/master/src/compiler.ts#L1316) that then leads us to the [declaration](https://github.com/AssemblyScript/assemblyscript/blob/master/src/compiler.ts#L1320) that we can resolve to a reflection object by its global name. The "Debug Failure" error seems to be from TypeScript, not exactly sure where. Would help to have a stack trace running from source.
